vsp/vsp.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2023 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package vsp
+
+import (
+	"context"
+	"net"
+
+	"decred.org/dcrwallet/v3/internal/vsp"
+	"github.com/decred/dcrd/dcrutil/v4"
+)
+
+// NewClient creats a new vsp client that can be used to purchase tickets,
+// monitor the vsp fee, and set vote preferences.
+func NewClient(wf vsp.WalletFetcher, vspHost, vspPubKey string, feeAcct, changeAcct uint32,
```

Comment:
We lose the benefits of the internal package when we use an internal interface in public API.  A change to the internal interface could unwittingly also modify public API.

The interface is gnarly (I assume because you want to pass in an implementation that talks to the wallet over RPC), but I feel it's better if we define it in this package.